src/Host.tsx

```diff
@@ -44,8 +44,36 @@ export const Host = ({ children, style }: IHostProps): JSX.Element => {
     }
   }, []);
 
-  const mount = (children: React.ReactNode): number => {
-    const key = Date.now();
+  //Keep track of all the keys we've generated to ensure uniqueness
+  let usedKeys: Array<string> = [];
+
+  //Makes a magical random key
+  const keyGenerator = (): string => {
```

Comment:
Can we move this function into a separated file? best would be `utils/file_name.ts`

Reply:
> Can we move this function into a separated file? best would be `utils/file_name.ts`

Yep makes total sense!

Also, question: I wondered whether the usedKeys array should really be stored in a ref, otherwise on every re-render of the Host, we're going to reset that array surely?

Reply:
Okay better then, we could export it under hooks/useGeneratedKey.ts where it keeps the ref there for the usedKeys and have the logic there

Reply:
OK I shall work on that :) writing in TypeScript is fairly new to me, so bear with!